[BUG] test_partitioned_parquet_write[PartitionWriteMode.Dynamic] failed in databricks runtimes

**Describe the bug**
introduced in https://github.com/NVIDIA/spark-rapids/pull/7653

`FAILED ../../src/main/python/parquet_write_test.py::test_partitioned_parquet_write[PartitionWriteMode.Dynamic][IGNORE_ORDER({'local': True})]` failed,
```
Part of the plan is not columnar class org.apache.spark.sql.execution.command.DataWritingCommandExec
```

```

[2023-02-05T11:42:48.050Z] E               pyspark.sql.utils.IllegalArgumentException: Part of the plan is not columnar class org.apache.spark.sql.execution.command.DataWritingCommandExec

[2023-02-05T11:42:48.050Z] E               Execute InsertIntoHiveTable `default`.`tmp_table_gw1_1094026942_2`, org.apache.hadoop.hive.ql.io.parquet.serde.ParquetHiveSerDe, [type=None], true, false, [make, model, year, comment, type]

[2023-02-05T11:42:48.050Z] E               +- GpuColumnarToRow false

[2023-02-05T11:42:48.050Z] E                  +- GpuProject [make#1430070, model#1430071, year#1430072, cast(type#1430073 as string) AS comment#1430080, cast(comment#1430074 as string) AS type#1430081], true

[2023-02-05T11:42:48.050Z] E                     +- GpuFileGpuScan parquet default.tmp_table_gw1_1094026942_0[make#1430070,model#1430071,year#1430072,type#1430073,comment#1430074] Batched: true, DataFilters: [], Format: Parquet, Location: InMemoryFileIndex[file:/home/ubuntu/spark-rapids/integration_tests/target/run_dir-20230205094731-..., PartitionFilters: [], PushedFilters: [], ReadSchema: struct<make:string,model:string,year:int,type:string,comment:string>
```

**Steps/Code to reproduce bug**
run parquet_write_test.py::test_partitioned_parquet_write in databricks ENV

**Expected behavior**
pass the test

--- a/integration_tests/src/main/python/parquet_write_test.py
+++ b/integration_tests/src/main/python/parquet_write_test.py
@@ -20,7 +20,7 @@ from data_gen import *
 from enum import Enum
 from marks import *
 from pyspark.sql.types import *
-from spark_session import with_cpu_session, with_gpu_session, is_before_spark_330, is_before_spark_320
+from spark_session import with_cpu_session, with_gpu_session, is_before_spark_330, is_before_spark_320, is_databricks_runtime
 import pyspark.sql.functions as f
 import pyspark.sql.utils
 import random
@@ -573,9 +573,12 @@ def test_write_empty_data_single_writer(spark_tmp_path):
 PartitionWriteMode = Enum('PartitionWriteMode', ['Static', 'Dynamic'])
 
 
+@pytest.mark.skipif(is_databricks_runtime(),
+                    reason="On Databricks, Hive partitioned SQL writes are routed through InsertIntoHiveTable; "
+                           "GpuInsertIntoHiveTable does not support Parquet writes.")
 @ignore_order(local=True)
 @pytest.mark.parametrize('mode', [PartitionWriteMode.Static, PartitionWriteMode.Dynamic])
-def test_partitioned_parquet_write(mode, spark_tmp_table_factory):
+def test_partitioned_sql_parquet_write(mode, spark_tmp_table_factory):
 
     def create_input_table(spark):
         tmp_input = spark_tmp_table_factory.get()
@@ -625,4 +628,38 @@ def test_partitioned_parquet_write(mode, spark_tmp_table_factory):
     assert_gpu_and_cpu_sql_writes_are_equal_collect(
         spark_tmp_table_factory, write_partitions,
         conf={"hive.exec.dynamic.partition.mode": "nonstrict"}
+    )
+
+
+@ignore_order(local=True)
+def test_dynamic_partitioned_parquet_write(spark_tmp_table_factory, spark_tmp_path):
+
+    def create_input_table(spark):
+        tmp_input = spark_tmp_table_factory.get()
+        spark.sql("CREATE TABLE " + tmp_input +
+                  " (make STRING, model STRING, year INT, type STRING, comment STRING)" +
+                  " STORED AS PARQUET")
+        spark.sql("INSERT INTO TABLE " + tmp_input + " VALUES " +
+                  "('Ford',   'F-150',       2020, 'ICE',      'Popular' ),"
+                  "('GMC',    'Sierra 1500', 1997, 'ICE',      'Older'),"
+                  "('Chevy',  'D-Max',       2015, 'ICE',      'Isuzu?' ),"
+                  "('Tesla',  'CyberTruck',  2025, 'Electric', 'BladeRunner'),"
+                  "('Rivian', 'R1T',         2022, 'Electric', 'Heavy'),"
+                  "('Jeep',   'Gladiator',   2024, 'Hybrid',   'Upcoming')")
+        return tmp_input
+
+    input_table_name = with_cpu_session(create_input_table)
+    base_output_path = spark_tmp_path + "/PARQUET_DYN_WRITE"
+
+    def write_partitions(spark, table_path):
+        input_df = spark.sql("SELECT * FROM {}".format(input_table_name))
+        input_df.write.mode("overwrite").partitionBy("type").parquet(table_path)
+        # Second write triggers the actual overwrite.
+        input_df.write.mode("overwrite").partitionBy("type").parquet(table_path)
+
+    assert_gpu_and_cpu_writes_are_equal_collect(
+        lambda spark, path: write_partitions(spark, path),
+        lambda spark, path: spark.read.parquet(path),
+        base_output_path,
+        conf={}
     )

